overlord/devicestate/firstboot_test.go

```diff
@@ -445,9 +442,18 @@ func (s *FirstBootTestSuite) TestPopulateFromSeedHappy(c *C) {
 func (s *FirstBootTestSuite) TestPopulateFromSeedMissingBootloader(c *C) {
 	chg := s.makeBecomeOpertionalChange(c)
 	st := s.overlord.State()
+
+	// run the change, we cannot use s.overlord.Settle() here as each
+	// "DeviceManager.Ensure()" will call ensureSeedYaml which will
+	// create a new "seed" change so things never settle down
+	s.overlord.Loop()
+	defer s.overlord.Stop()
+	for !chg.IsReady() {
+		// nothing
+	}
+
```

Comment:
I have not yet figured out why this affects the test, I mean, I understand why settle() never finishes, but I have not tracked down why this was working before because we call ensureSeedYaml a lot.